bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -623,13 +655,32 @@ func (r *KubeadmConfigReconciler) joinControlplane(ctx context.Context, scope *S
 			KubeadmVerbosity:     verbosityFlag,
 			UseExperimentalRetry: scope.Config.Spec.UseExperimentalRetryJoin,
 		},
-	})
+	}
+
+	var bootstrapJoinData []byte
+	switch scope.Config.Spec.Format {
+	case bootstrapv1.Ignition:
+		ign, _, err := ignition.NewJoinControlPlane(&ignition.ControlPlaneJoinInput{
+			ControlPlaneJoinInput: controlPlaneJoinInput,
+			Ignition:              scope.Config.Spec.Ignition,
+		})
+
+		if err != nil {
+			scope.Error(err, "Failed to generate Ignition user data for join control plane")
+			return ctrl.Result{}, err
+		}
+
+		bootstrapJoinData = ign
```

Comment:
```suggestion
		bootstrapJoinData, _, err = ignition.NewJoinControlPlane(&ignition.ControlPlaneJoinInput{
			ControlPlaneJoinInput: controlPlaneJoinInput,
			Ignition:              scope.Config.Spec.Ignition,
		})
```
(same as above)

Reply:
Fixed and added a test to ensure correct handling.

Reply:
Ditto